astropy/modeling/fitting.py

```diff
@@ -57,6 +57,72 @@
 OPTIMIZERS = [Simplex, SLSQP]
 
 
+class Covariance():
+    """Class for covariance matrix calculated by fitter. """
+
+    def __init__(self, cov_matrix, param_names):
+        self.cov_matrix = cov_matrix
+        self.param_names = param_names
+
+    def pprint(self, max_lines, round_val):
+        # Print and label lower triangle of covariance matrix
+        # Print rows for params up to `max_lines`, round floats to 'round_val'
+        longest_name = max([len(x) for x in self.param_names])
+        ret_str = 'parameter variances / covariances \n'
+        for i, row in enumerate(self.cov_matrix):
+            if i <= max_lines:
+                param = self.param_names[i]
+                s = param + ' ' * (longest_name - len(param)) + ' |'
+                ret_str += s + ', '.join([str(np.round(x, round_val))
+                           for x in row[0:i+1]]) + '\n'
+            else:
+                ret_str += '...'
+        return(ret_str)
+        
+    def __repr__(self):
+        return(self.pprint(max_lines=10, round_val=3))
+
+    def __getitem__(self, params):
+        # index covariance matrix by parameter names or indicies
+        if all(isinstance(item, str) for item in params):
+            i1, i2 = self.param_names.index(params[0]), self.param_names.index(params[1])
+        elif all(isinstance(item, int) for item in params):
+            i1, i2 = params
+        else:
+            raise TypeError('Covariance must be indexed by two parameter names, or integer indicies.')
+        return(self.cov_matrix[i1][i2])
+
```

Comment:
Fixed and tied parameters are excluded from the list of fitted parameters. For example, a gaussian with one fixed parameter will have a cov_matrix of shape (2, 2), instead of (3, 3).

Reply:
@perrygreenfield that should be how it works, but since i'm currently using model.param_names, which contains all 
 parameters including fixed + tied parameters, it doesn't work in it's current state. I'll change that.

Reply:
I think this does work as expected now.